Tpetra::RowMatrixTransposer: Build warning in non-ETI build

## Bug Report
@trilinos/tpetra 

Reported by Sierra's Trilinos integration build.

### Description

```
In file included from .../include/Tpetra_RowMatrixTransposer.hpp:2:0,
                 from .../include/TpetraExt_MatrixMatrix_def.hpp:50,
                 from .../include/TpetraExt_MatrixMatrix.hpp:2,
                 from ...,
.../include/Tpetra_RowMatrixTransposer_def.hpp:57:1: error: 'Kokkos::Impl::ViewCtorProp, std::allocator >, Kokkos::Impl::WithoutInitializing_t> Tpetra::{anonymous}::view_no_init(const string&)' defined but not used [-Werror=unused-function]
 view_no_init (const std::string& label) ->
 ^~~~~~~~~~~~
cc1plus: all warnings being treated as errors
```

The issue appears to be that inlined nontemplated functions in a header file issue these warnings, even if the function is used in templated functions in the file.  This warning only appears in a non-ETI build.

### Steps to Reproduce

GCC 7.2.0, OpenMPI 1.10.2, warnings as errors, ETI off.

--- a/packages/tpetra/core/src/Tpetra_RowMatrixTransposer_def.hpp
+++ b/packages/tpetra/core/src/Tpetra_RowMatrixTransposer_def.hpp
@@ -51,17 +51,6 @@
 
 namespace Tpetra {
 
-namespace { // (anonymous)
-
-auto
-view_no_init (const std::string& label) ->
-  decltype (Kokkos::view_alloc (label, Kokkos::WithoutInitializing))
-{
-  return Kokkos::view_alloc (label, Kokkos::WithoutInitializing);
-}
-
-} // namespace (anonymous)
-
 template<class Scalar,
          class LocalOrdinal,
          class GlobalOrdinal,
@@ -216,15 +205,19 @@ createTransposeLocal (const Teuchos::RCP<Teuchos::ParameterList>& params)
       }
     });
 
+  using Kokkos::view_alloc;
+  using Kokkos::WithoutInitializing;
   row_map_type t_offsets
-    (view_no_init ("transpose ptr"), lclNumCols + 1);
+    (view_alloc ("transpose ptr", WithoutInitializing), lclNumCols + 1);
 
   // TODO (mfh 10 Jul 2019): This returns the sum of all counts,
   // which could be useful for checking nnz.
   (void) Details::computeOffsetsFromCounts (t_offsets, t_counts);
 
-  index_type t_cols (view_no_init ("transpose lcl ind"), nnz);
-  values_type t_vals (view_no_init ("transpose val"), nnz);
+  index_type t_cols
+    (view_alloc ("transpose lcl ind", WithoutInitializing), nnz);
+  values_type t_vals
+    (view_alloc ("transpose val", WithoutInitializing), nnz);
   Kokkos::parallel_for
     ("Compute local transpose",
      range_type (0, lclNumRows),

